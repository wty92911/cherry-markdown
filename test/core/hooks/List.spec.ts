import List from '../../../src/core/hooks/List';
import * as md5 from 'md5';

const cases = []
cases[0] = `
- 1
- 2
  - 2.1
  - 2.2
- 3
  + 3.1
- 4
  * 4.2
`;

cases[1] = `
- 1
  - 2
       - 2.1
    - 2.2
 - 3
          + 3.1
 - 4
* 4.2
`;

cases[2] = `
- 1
1. test

- 1.1
   - 1.1.2
       - blank
  - 1.2
 - 2
          + blank
      - 2.1
	* 2.2
 1. test
   2. 2
`;

cases[3] = `
1. test
	2. test
1. test
   ä¸€. test
   1. test
   
   
   a. test
- test
`;

const listHook = new List({ config: {}});
Object.defineProperty(listHook, '$engine', {
  value: { md5 },
});

describe('core/hooks/list', () => {
  it('checklist replace', () => {
    cases.forEach((item) => {
      expect(listHook.toHtml(item, (text) => ({ html: text }))).toMatchSnapshot();
    });
  });
});
