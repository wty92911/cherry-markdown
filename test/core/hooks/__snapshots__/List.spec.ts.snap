// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core/hooks/list list hook 1`] = `"<ul data-lines=\\"8\\" data-sign=\\"0ee90839df66d748d6faefc70677e3ee\\" class=\\"cherry-list__default\\"><li><p>1</p></li><li><p>2</p><ul class=\\"cherry-list__default\\"><li><p>2.1</p></li><li><p>2.2</p></li></ul></li><li><p>3</p><ul class=\\"cherry-list__circle\\"><li><p>3.1</p></li></ul></li><li><p>4</p><ul class=\\"cherry-list__square\\"><li><p>4.2</p></li></ul></li></ul>"`;

exports[`core/hooks/list list hook 2`] = `"<ul data-lines=\\"8\\" data-sign=\\"517b0da7dd7b1b616ab99665b9d280eb\\" class=\\"cherry-list__default\\"><li><p>1</p><ul class=\\"cherry-list__default\\"><li><p>2</p><ul class=\\"cherry-list__default\\"><li><p>2.1</p></li><li><p>2.2</p></li></ul></li></ul></li><li><p>3<br>3.1</p></li><li><p>4</p></li><li><p>4.2</p></li></ul>"`;

exports[`core/hooks/list list hook 3`] = `"<ul data-lines=\\"1\\" data-sign=\\"567632d2786fbb27c9e5df62189c44a7\\" class=\\"cherry-list__default\\"><li><p>1</p></li></ul><ol data-lines=\\"2\\" data-sign=\\"567632d2786fbb27c9e5df62189c44a7\\" class=\\"cherry-list__default\\"><li><p>test<br></p></li></ol><ul data-lines=\\"8\\" data-sign=\\"567632d2786fbb27c9e5df62189c44a7\\" class=\\"cherry-list__default\\"><li><p>1.1</p><ul class=\\"cherry-list__default\\"><li><p>1.1.2</p><ul class=\\"cherry-list__default\\"><li><p>blank</p></li></ul></li><li><p>1.2</p></li></ul></li><li><p>2<br>blank</p><ul class=\\"cherry-list__default\\"><li><p>2.1</p></li><li><p>2.2</p></li></ul></li></ul><ol data-lines=\\"2\\" data-sign=\\"567632d2786fbb27c9e5df62189c44a7\\" class=\\"cherry-list__default\\"><li><p>test</p><ol class=\\"cherry-list__default\\"><li><p>2</p></li></ol></li></ol>"`;

exports[`core/hooks/list list hook 4`] = `"<ol data-lines=\\"8\\" data-sign=\\"a7f4c176bd29fce6eed2db95c7433f88\\" class=\\"cherry-list__default\\"><li><p>test</p><ol class=\\"cherry-list__default\\"><li><p>test</p></li></ol></li><li><p>test</p><ol class=\\"cherry-list__cjk-ideographic\\"><li><p>test</p></li><li><p>test<br><br></p></li><li><p>test</p></li></ol></li></ol><ul data-lines=\\"1\\" data-sign=\\"a7f4c176bd29fce6eed2db95c7433f88\\" class=\\"cherry-list__default\\"><li><p>test</p></li></ul>"`;

exports[`core/hooks/list list hook 5`] = `"<ul data-lines=\\"6\\" data-sign=\\"1352e6eb28a6f713e316fbe3733cdee7\\" class=\\"cherry-list__default\\"><li class=\\"check-list-item\\"><p><span class=\\"ch-icon ch-icon-square\\"></span> checklist 1</p></li><li><p>test</p><ul class=\\"cherry-list__default\\"><li class=\\"check-list-item\\"><p><span class=\\"ch-icon ch-icon-check\\"></span> checklist 2</p></li></ul></li><li class=\\"check-list-item\\"><p><span class=\\"ch-icon ch-icon-square\\"></span> checklist 3</p></li><li><p>test</p><ul class=\\"cherry-list__default\\"><li class=\\"check-list-item\\"><p><span class=\\"ch-icon ch-icon-square\\"></span> checklist 4</p></li></ul></li></ul>"`;
